unified-codes/.dist.env

```diff
@@ -0,0 +1,23 @@
+## For production builds, rename this file as .local.env.
```

Comment:
Isn't this the production version?

Reply:
Hm, maybe misunderstanding the Q, but by renaming as `.local.env`, this file will take precedence over `.env` (see https://nx.dev/react/guides/environment-variables#environment-variables). On rereading, I notice the react docs mention having to prepend with `NX`... but the angular docs (the ones I was reading) don't... everything seems to be working, however... so maybe just leave as is (@katherine-sussol?).

In case still not clear, this is how I was thinking:

- Debug: running things locally.
- Production: running things potentially on different hosts (may be local, depending on how we set it up, but may be mixed, e.g. keycloak hosted externally, or something!).
- Docker: somewhat ambiguous, because once we get everything working, docker will be production... but for now, keeping them separate!

It's all quite in flux, was just trying to find a neat way of separating things and keeping everything at the root level... but I think we will probably end up with just a single `.env` with everything local, and then the docker `.env` with everything on the docker network (i.e. at least to start with, the distributed system will be all on a single machine)?

Reply:
I see `dist` as production - it's the built / distributable version; the contents of this file looked like production values too. I think that `local` is useful to denote the local dev environment. you have nice values in the example file which goes through source control - called '.prod.env' or '.dist.env' or '.env' and a local version which doesn't live in source control - and which devs copy from the example and tailor to their own needs. My two cents anyway.

if there's any monkeying around to be done with filenames, can we get the build script to do the renaming.

Reply:
I prefer the way you've done it @wlthomson  r.e. NX prefix (especially given it works without the prefix), much neater.
R.e. `.env` files in general I like the idea of reserving local for local use that we modify on our machines (it just feels more intuitive in a file with local in its name). Maybe we could keep the current files, but for production, overwrite the `.env` file with the contents of `.dist.env`, and not have any .local file at all? (done via build script of course 😄 ).

Also agree that eventually maybe docker file isn't needed and will just be merged into `dist.env`, but good to wait until we can test a deployment before discarding one or the other.

Reply:
@mark-prins Maybe I was unclear again 😨! That file is indeed the production file, renaming as `.local` just gives it priority over the other `.env` file (which is debug). Does that make sense?

Yeah, I agree, I initially had it so all files were prepended, e.g.: `.debug.env`, `.dist.env`... and the user would rename whichever one was appropriate to `.env`. Does that sound about right?

Reply:
ef03c25e3.